src/correction/cellbender_remove_background/helper.py

```diff
@@ -0,0 +1,137 @@
+    if 'genes' in d.keys():
+        d['id'] = d.pop('genes')
+
+    # For purely aesthetic purposes, rename "id" to "gene_id"
+    if 'id' in d.keys():
+        d['gene_id'] = d.pop('id')
+
+    # If genomes are empty, try to guess them based on gene_id
+    if 'genome' in d.keys():
+        if np.array([s.decode() == '' for s in d['genome']]).all():
+            if '_' in d['gene_id'][0].decode():
+                print('Genome field blank, so attempting to guess genomes based on gene_id prefixes')
+                d['genome'] = np.array([s.decode().split('_')[0] for s in d['gene_id']], dtype=str)
+
+    # Add other information to the anndata object in the appropriate slot.
+    _fill_adata_slots_automatically(adata, d)
+
+    # Add a special additional field to .var if it exists.
+    if 'features_analyzed_inds' in adata.uns.keys():
+        adata.var['cellbender_analyzed'] = [True if (i in adata.uns['features_analyzed_inds'])
+                                            else False for i in range(adata.shape[1])]
+
+    if analyzed_barcodes_only:
+        for col in adata.obs.columns[adata.obs.columns.str.startswith('barcodes_analyzed')
+                                     | adata.obs.columns.str.startswith('barcode_indices')]:
+            try:
+                del adata.obs[col]
+            except Exception:
+                pass
+    else:
+        # Add a special additional field to .obs if all barcodes are included.
+        if 'barcodes_analyzed_inds' in adata.uns.keys():
+            adata.obs['cellbender_analyzed'] = [True if (i in adata.uns['barcodes_analyzed_inds'])
+                                                else False for i in range(adata.shape[0])]
+
+    return adata
+
+
+def dict_from_h5(file: str) -> Dict[str, np.ndarray]:
+    """Read in everything from an h5 file and put into a dictionary."""
+    d = {}
+    with tables.open_file(file) as f:
+        # read in everything
+        for array in f.walk_nodes("/", "Array"):
+            d[array.name] = array.read()
+    return d
+
+
+def _fill_adata_slots_automatically(adata, d):
+    """Add other information to the adata object in the appropriate slot."""
+
+    for key, value in d.items():
+        try:
+            if value is None:
+                continue
+            value = np.asarray(value)
+            if len(value.shape) == 0:
+                adata.uns[key] = value
+            elif value.shape[0] == adata.shape[0]:
+                if (len(value.shape) < 2) or (value.shape[1] < 2):
+                    adata.obs[key] = value
+                else:
+                    adata.obsm[key] = value
+            elif value.shape[0] == adata.shape[1]:
+                if value.dtype.name.startswith('bytes'):
+                    adata.var[key] = value.astype(str)
+                else:
+                    adata.var[key] = value
+            else:
+                adata.uns[key] = value
+        except Exception:
```

Comment:
I think this can be restricted to `ValueError`, `TypeError` and `KeyError`? A general exception also captures stuff like `MemoryError` and `SyntaxError`

```python3
try:
    ...
except (ValueError, TypeError, KeyError):
    ...
```